### PR TITLE
Misc doc changes

### PR DESCRIPTION
**Pull request by ronanstokes-db**
## Proposed changes

Miscellaneous doc changes for the coming release

## Types of changes

What types of changes does your code introduce to dbldatagen?
_Put an `x` in the boxes that apply_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Change to tutorials, tests or examples
- [x] Non code change (readme, images or other non-code assets)
- [x] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. 
If you're unsure about any of them, don't hesitate to ask. We're here to help! 
This is simply a reminder of what we are going to look for before merging your code._

- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] Submission does not reduce code coverage numbers
- [x] Submission does not increase alerts or messages from prospector / lint

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you 
did and what alternatives you considered, etc...

### CODE REVIEW

File: docs/source/textdata.rst [hunk 1]
```diff
@@ -154,6 +155,17 @@ It uses the following special chars:
...
     ========  ======================================
 
+In all other cases, the char itself is used.
+
+The setting of the ``escapeSpecialChars`` determines how templates generate data.
+
+If set to False, then the template ``r"\\dr_\\v"`` will generate the values ``"dr_0"`` ... ``"dr_999"`` when applied
+to the values zero to 999. This conforms to earlier implementations for backwards compatibility.
+
+If set to True, then the template ``r"dr_\\v"`` will generate the values ``"dr_0"`` ... ``"dr_999"``
```
Comment (howardwu-db) [inline docs/source/textdata.rst:165]: Shouldnt this look different than when set to False?

**Review by howardwu-db — CHANGES_REQUESTED** [review]

File: docs/source/textdata.rst [hunk 2]
```diff
@@ -165,20 +177,35 @@ It uses the following special chars:
...
+
+The use of the `values` option and the `template` option with a `TemplateGenerator` instance allow for generation of
+data when the range of possible values is known.
+
+But what about scenarios when the list of data is read from a different table or some other form of lookup?
+
+As the output of the data generation `build()` method is a regular PySpark DataFrame, it is possible to join the
+generated data with other data sources to generate the required data.
 
+In these cases, the generator can be specified to produce lookup keys that can be used to join with the
```
Comment (howardwu-db) [inline docs/source/textdata.rst:209]: Can we get an example of how to do this?

Comment (ronanstokes-db) [reply docs/source/textdata.rst:209]: lets do this in a later checkin

File: docs/source/textdata.rst (hunk 1 above)
Comment (ronanstokes-db) [reply docs/source/textdata.rst]: reworked this section

File: docs/source/textdata.rst (hunk 2 above)
Comment (ronanstokes-db) [reply docs/source/textdata.rst:257]: added

**Review by howardwu-db — APPROVED** [review]

**Pull request merged**